SessionPoolTest.testMaintainerKeepsWriteProportion is flaky

https://github.com/googleapis/java-spanner/pull/259/checks?check_run_id=761572821

```
[ERROR] Tests run: 68, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.822 s <<< FAILURE! - in com.google.cloud.spanner.SessionPoolTest
[ERROR] com.google.cloud.spanner.SessionPoolTest.testMaintainerKeepsWriteProportion[min sessions = 0]  Time elapsed: 0.047 s  <<< FAILURE!
com.google.common.truth.ComparisonFailureWithFacts: 
expected: 5
but was : 4
	at com.google.cloud.spanner.SessionPoolTest.testMaintainerKeepsWriteProportion(SessionPoolTest.java:1064)
```

--- a/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java
+++ b/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java
@@ -1324,6 +1324,13 @@ final class SessionPool {
   }
 
   @VisibleForTesting
+  int getNumberOfWriteSessionsInPool() {
+    synchronized (lock) {
+      return writePreparedSessions.size() + numSessionsBeingPrepared;
+    }
+  }
+
+  @VisibleForTesting
   int getNumberOfSessionsBeingCreated() {
     synchronized (lock) {
       return numSessionsBeingCreated;

--- a/google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolTest.java
+++ b/google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolTest.java
@@ -1058,9 +1058,7 @@ public class SessionPoolTest extends BaseSessionPoolTest {
     verify(session, times(options.getMinSessions())).singleUse(any(TimestampBound.class));
     // Verify that all sessions are still in the pool, and that the write fraction is maintained.
     assertThat(pool.getNumberOfSessionsInPool()).isEqualTo(options.getMinSessions());
-    assertThat(
-            pool.getNumberOfAvailableWritePreparedSessions()
-                + pool.getNumberOfSessionsBeingPrepared())
+    assertThat(pool.getNumberOfWriteSessionsInPool())
         .isEqualTo(
             (int) Math.ceil(pool.getNumberOfSessionsInPool() * options.getWriteSessionsFraction()));
 

